enclave-bridge/src/lib.rs

```diff
@@ -235,12 +243,55 @@ pub mod pallet {
 
 			Ok(().into())
 		}
+
+		/// Update shard config
+		/// To be respected by L2 instances after `enactment_delay` parentchain blocks
+		/// If no previous config exists, the `enactment_delay` parameter will be ignored
+		/// and the `shard_config` will be active immediately
+		#[pallet::call_index(5)]
+		#[pallet::weight((1000, DispatchClass::Normal, Pays::No))]
+		pub fn update_shard_config(
+			origin: OriginFor<T>,
+			shard: ShardIdentifier,
+			shard_config: ShardConfig<T::AccountId>,
+			enactment_delay: T::BlockNumber,
+		) -> DispatchResultWithPostInfo {
+			let sender = ensure_signed(origin)?;
```

Comment:
With the separation of concerns, I believe that the semantics of this dispatchable should also be changed. This dispatchable should only be allowed to update existing shards, if the shard does not exist, it should return an error. We can have a separate dispatchable for creating shards.